compiler/src/lntoamm/Statement.ts

```diff
@@ -43,106 +49,76 @@ class Statement {
     return false
   }
 
-  static assignmentsHasObjectLiteral(assignmentsAst: any) { // TODO: Remove ANTLR
-    return Statement.assignablesHasObjectLiteral(assignmentsAst.assignables())
+  static assignmentsHasObjectLiteral(assignmentsAst: LPNode) {
+    return Statement.assignablesHasObjectLiteral(assignmentsAst.get('assignables'))
   }
 
   hasObjectLiteral() {
     const s = this.statementAst
-    if (s.declarations()) {
-      const d = s.declarations().constdeclaration() || s.declarations().letdeclaration()
-      return Statement.assignablesHasObjectLiteral(d.assignables())
+    if (s.has('declarations')) {
+      const d = s.get('declarations').has('constdeclaration') ?
+        s.get('declarations').get('constdeclaration') :
+        s.get('declarations').get('letdeclaration')
+      return Statement.assignablesHasObjectLiteral(d.get('assignables'))
     }
-    if (s.assignments()) return Statement.assignmentsHasObjectLiteral(s.assignments())
-    if (s.assignables()) return Statement.assignablesHasObjectLiteral(s.assignables())
-    if (s.exits() && s.exits().assignables()) return Statement.assignablesHasObjectLiteral(
-      s.exits().assignables()
-    )
-    if (s.emits() && s.emits().assignables()) return Statement.assignablesHasObjectLiteral(
-      s.emits().assignables()
-    )
-    // TODO: Cover conditionals
-    return false
-  }
-
-  static isCallPure(callAst: any, scope: Scope) { // TODO: Migrate off ANTLR
```

Comment:
I deleted this because while checking the purity of a function is desired so we know whether or not to automatically un-parallelize Array methods, the prior implementation didn't work and we weren't using it, so we should just add it back from a first-principles perspective in the future.